moveit_ros/moveit_servo/include/moveit_servo/servo_parameters.cpp

```diff
@@ -0,0 +1,278 @@
+
+  // Incoming Joint State properties
+  error |= !declareAndGetParam<std::string>(ns + ".joint_topic", parameters->joint_topic, node, logger);
+  error |= !declareAndGetParam<double>(ns + ".low_pass_filter_coeff", parameters->low_pass_filter_coeff, node, logger);
+
+  // MoveIt properties
+  error |= !declareAndGetParam<std::string>(ns + ".move_group_name", parameters->move_group_name, node, logger);
+  error |= !declareAndGetParam<std::string>(ns + ".planning_frame", parameters->planning_frame, node, logger);
+
+  // Stopping behaviour
+  error |= !declareAndGetParam<double>(ns + ".incoming_command_timeout", parameters->incoming_command_timeout, node, logger);
+  error |= !declareAndGetParam<int>(ns + ".num_outgoing_halt_msgs_to_publish", parameters->num_outgoing_halt_msgs_to_publish,
+                                   node, logger);
+
+  // Configure handling of singularities and joint limits
+  error |=
+      !declareAndGetParam<double>(ns + ".lower_singularity_threshold", parameters->lower_singularity_threshold, node, logger);
+  error |= !declareAndGetParam<double>(ns + ".hard_stop_singularity_threshold", parameters->hard_stop_singularity_threshold,
+                                      node, logger);
+  error |= !declareAndGetParam<double>(ns + ".joint_limit_margin", parameters->joint_limit_margin, node, logger);
+
+  // Collision checking
+  error |= !declareAndGetParam<bool>(ns + ".check_collisions", parameters->check_collisions, node, logger);
+  error |= !declareAndGetParam<double>(ns + ".collision_check_rate", parameters->collision_check_rate, node, logger);
+  error |= !declareAndGetParam<std::string>(ns + ".collision_check_type", parameters->collision_check_type, node, logger);
+  error |= !declareAndGetParam<double>(ns + ".self_collision_proximity_threshold",
+                                      parameters->self_collision_proximity_threshold, node, logger);
+  error |= !declareAndGetParam<double>(ns + ".scene_collision_proximity_threshold",
+                                      parameters->scene_collision_proximity_threshold, node, logger);
+  error |= !declareAndGetParam<double>(ns + ".collision_distance_safety_factor", parameters->collision_distance_safety_factor,
+                                      node, logger);
+  error |= !declareAndGetParam<double>(ns + ".min_allowable_collision_distance", parameters->min_allowable_collision_distance,
+                                      node, logger);
+
+  // Only continue if all parameters were found
+  if (error)
+  {
+    RCLCPP_ERROR_STREAM(logger, "One or more Servo parameters missing, check YAML file before proceeding");
+    return false;
+  }
+
+  // Begin input checking
+  if (parameters->publish_period <= 0.)
+  {
+    RCLCPP_WARN(logger, "Parameter 'publish_period' should be "
+                        "greater than zero. Check yaml file.");
+    return false;
+  }
+  if (parameters->num_outgoing_halt_msgs_to_publish < 0)
```

Comment:
should be <=, I think

Reply:
FYI only the syntax was changed here, so this logic exists in ROS1 [here](https://github.com/ros-planning/moveit/blob/589f83a132da174991b233931d3ca7f31ab1cba4/moveit_ros/moveit_servo/src/servo.cpp#L170)

Reply:
IMO that's not significant enough to merit a PR in ROS1, but might as well fix it here.

(Feel free to make a ROS1 PR if you want to, but we already have like 2-3 open PRs that need reviewing)

Reply:
Maybe you could sneak it into this PR you already started

https://github.com/ros-planning/moveit/pull/2249

Reply:
[This line](https://github.com/ros-planning/moveit/blob/ed3e196d28d670fb154af3cb2866227448b3ad32/moveit_ros/moveit_servo/src/servo_calcs.cpp#L309-L310) in `servo_calcs.cpp` indicates that `num_outgoing_halt_msgs_to_publish` can be 0, do we want to keep it?

`// num_outgoing_halt_msgs_to_publish == 0 signifies that we should keep republishing forever.
  if (!have_nonzero_command_ && (parameters_.num_outgoing_halt_msgs_to_publish != 0) &&`

Reply:
yep, good catch